pkg/resources/alerting/base.go

```diff
@@ -0,0 +1,387 @@
+package alerting
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/rancher/opni/pkg/alerting/shared"
+	"github.com/rancher/opni/pkg/resources"
+	"github.com/rancher/opni/pkg/util"
+	"github.com/samber/lo"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const dataMountPath = "/var/lib"
+
+func alertingLabels() map[string]string {
+	return map[string]string{
+		"app.kubernetes.io/name": "opni-alerting",
+	}
+}
+
+func (r *Reconciler) alerting() []resources.Resource {
+	labelWithAlert := func(label map[string]string) map[string]string {
+		label["app.kubernetes.io/name"] = "opni-alerting"
+		return label
+	}
+	labelWithNode := func(label map[string]string) map[string]string {
+		label["app.alerting.type"] = "node"
+		return label
+	}
+	labelWithCluster := func(label map[string]string) map[string]string {
+		label["app.alerting.node"] = "controller"
+		return label
+	}
+	publicNodeLabels := labelWithNode(labelWithAlert(map[string]string{}))
+	publicNodeSvcLabels := publicNodeLabels
+	publicControllerLabels := labelWithCluster(labelWithAlert(map[string]string{}))
+	publicControllerSvcLabels := publicControllerLabels
+	requiredVolumes := []corev1.Volume{
+		{
+			Name: "opni-alertmanager-data",
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: "opni-alertmanager-data",
+					ReadOnly:  false,
+				},
+			},
+		},
+	}
```

Comment:
What happens if the user has multiple storage classes and they want to select one?  What if they don't have any storage classes?

Reply:
that's good point, I'll look into adding a storage class to the alerting cluster crd

Reply:
Ended up using a more generic approach of overloading the PVC template Spec

Reply:
Using the PVCSource from opster seems like a better approach here.